dev/com.ibm.ws.security.openidconnect.clients.common/src/com/ibm/ws/security/openidconnect/client/jose4j/util/Jose4jUtil.java

```diff
@@ -429,4 +428,31 @@ ProviderAuthenticationResult checkForReusedJwt(ConvergedClientConfig clientConfi
         return null;
     }
 
+    /**
+     * Verifies that the JWT is either a JWS or a JWE, depending on the client configuration, and does simple parsing of the
+     * token to ensure it is formatted correctly. If the token is a JWE, this decrypts and extracts the JWS payload from the
+     * token.
+     */
+    public JwtContext validateJwtStructureAndGetContext(String logoutTokenString, ConvergedClientConfig clientConfig) throws Exception {
```

Comment:
Since this can handle the token during authentication, the logoutTokenString should be renamed.

Reply:
Ah, yep. I think I probably originally wrote this in the LogoutTokenValidator class and then moved it over. I'll get that renamed.